src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/TypedArray.cs

```diff
@@ -0,0 +1,196 @@
+
+                if (exception != 0)
+                    throw new JSException((string)res);
+
+            }
+        }
+
+        private U? UnBoxValue(object jsValue)
+        {
+            if (jsValue == null)
+                return null;
+
+            Type type = jsValue.GetType();
+            return (U)Convert.ChangeType(jsValue, typeof(U));
+        }
+
+        public U[] ToArray()
+        {
+            object res = Interop.Runtime.TypedArrayToArray(JSHandle, out int exception);
+
+            if (exception != 0)
+                throw new JSException((string)res);
+            return (U[])res;
+        }
+
+        public static unsafe T From(ReadOnlySpan<U> span)
+        {
+            // source has to be instantiated.
+            if (span == null)
+            {
+                throw new System.ArgumentException($"Invalid argument: {nameof(span)} can not be null.");
+            }
+
+            TypedArrayTypeCode type = (TypedArrayTypeCode)Type.GetTypeCode(typeof(U));
+            // Special case for Uint8ClampedArray, a clamped array which represents an array of 8-bit unsigned integers clamped to 0-255;
+            if (type == TypedArrayTypeCode.Uint8Array && typeof(T) == typeof(Uint8ClampedArray))
+                type = TypedArrayTypeCode.Uint8ClampedArray;  // This is only passed to the JavaScript side so it knows it will be a Uint8ClampedArray
+
+            ReadOnlySpan<byte> bytes = MemoryMarshal.AsBytes(span);
+            fixed (byte* ptr = bytes)
+            {
+                object res = Interop.Runtime.TypedArrayFrom((int)ptr, 0, span.Length, Marshal.SizeOf<U>(), (int)type, out int exception);
+                if (exception != 0)
+                    throw new JSException((string)res);
+                return (T)res;
+            }
+
+        }
+
+        public unsafe int CopyTo(Span<U> span)
+        {
+            ReadOnlySpan<byte> bytes = MemoryMarshal.AsBytes(span);
+            fixed (byte* ptr = bytes)
+            {
+                object res = Interop.Runtime.TypedArrayCopyTo(JSHandle, (int)ptr, 0, span.Length, Marshal.SizeOf<U>(), out int exception);
+                if (exception != 0)
+                    throw new JSException((string)res);
+                return (int)res / Marshal.SizeOf<U>();
+            }
+        }
+
+        public unsafe int CopyFrom(ReadOnlySpan<U> span)
+        {
+            // source has to be instantiated.
+            if (span == null || span.Length == 0)
+            {
+                throw new System.ArgumentException($"Invalid argument: {nameof(span)} can not be null and must have a length");
+            }
+
+            ReadOnlySpan<byte> bytes = MemoryMarshal.AsBytes(span);
+            fixed (byte* ptr = bytes)
+            {
+                object res = Interop.Runtime.TypedArrayCopyFrom(JSHandle, (int)ptr, 0, span.Length, Marshal.SizeOf<U>(), out int exception);
```

Comment:
`Marshal.SizeOf` does not return the right size for non-blitable types here, and it is somewhat expensive API (on CoreCLR at least).

This should use `sizeof(U)` or `Unsafe.SizeOf<T>`.

Reply:
Addressed in recent commits